src/cares_wrap.cc

```diff
@@ -490,15 +485,26 @@ void ChannelWrap::Setup() {
   }
 
   library_inited_ = true;
+}
 
-  /* Initialize the timeout timer. The timer won't be started until the */
-  /* first socket is opened. */
-  CleanupTimer();
-  timer_handle_ = new uv_timer_t();
-  timer_handle_->data = static_cast<void*>(this);
-  uv_timer_init(env()->event_loop(), timer_handle_);
+void ChannelWrap::StartTimer() {
+  if (timer_handle_ == nullptr) {
```

Comment:
`if (timer_handle_ != nullptr) return;` - saves a level of indent.

Reply:
It's a bit more complex if you look below. We also check if the handle is active in an `else if`. (We need to be able to call `uv_timer_start` to restart the timer if it expired but the handle still exists.)